src/tests/system/CMakeLists.txt

```diff
@@ -0,0 +1,5 @@
+add_executable(test_sleep test_sleep.f90)
+target_link_libraries(test_sleep fortran_stdlib)
+
+add_test(NAME Sleep COMMAND $<TARGET_FILE:test_sleep> 1234)
```

Comment:
This sleeps for over 1s, so it makes all the tests hang to wait for this one to finish. See the timings:
```
$ ctest 
Test project /home/ondrej/repos/stdlib
      Start  1: AlwaysSkip
      Start  2: AlwaysFail
      Start  3: ASCII
      Start  4: loadtxt
 1/11 Test  #1: AlwaysSkip .......................***Skipped   0.01 sec
 2/11 Test  #2: AlwaysFail .......................   Passed    0.01 sec
 3/11 Test  #3: ASCII ............................   Passed    0.01 sec
 4/11 Test  #4: loadtxt ..........................   Passed    0.01 sec
      Start  5: savetxt
      Start  6: loadtxt_qp
      Start  7: savetxt_qp
      Start  8: open
 5/11 Test  #8: open .............................   Passed    0.00 sec
 6/11 Test  #7: savetxt_qp .......................   Passed    0.00 sec
 7/11 Test  #6: loadtxt_qp .......................   Passed    0.00 sec
 8/11 Test  #5: savetxt ..........................   Passed    0.00 sec
      Start  9: parse_mode
      Start 10: OPTVAL
      Start 11: Sleep
 9/11 Test #10: OPTVAL ...........................   Passed    0.10 sec
10/11 Test  #9: parse_mode .......................   Passed    0.10 sec
11/11 Test #11: Sleep ............................   Passed    1.24 sec

100% tests passed, 0 tests failed out of 11

Label Time Summary:
quadruple_precision    =   0.00 sec*proc (2 tests)

Total Test time (real) =   1.26 sec

The following tests did not run:
	  1 - AlwaysSkip (Skipped)
```
I would lower the sleeping amount.